kernel/src/syscall.rs

```diff
@@ -437,6 +440,11 @@ pub enum SyscallReturn {
     /// Generic success case, with an additional 32-bit and 64-bit data field
     SuccessU32U64(u32, u64),
 
+    /// Generic success case, with an additional pointer with metadata
+    /// On CHERI, this grants authority.
+    /// Access to this return is therefore privileged.
+    SuccessPtr(MetaPtr),
+
```

Comment:
This should be reflected in an update to TRD 104. Even if it isn't a breaking change, we should enumerate this variant and show onto which `SyscallReturnVariant` it maps.

Reply:
Isn't passing a pointer back to userspace via a command a very significant change?

Reply:
So, I think this crops up in two places:

The return value from allow (which I think _did_ previously return the last allow-ed thing?)
And sbrk, which did previously return an address, but on CHERI needed to return a MetaPtr to allow access to the new heap. This is actually new.

Possibly this is where I broke libtock-c, because these codes have changed. On the CHERI fork of libtock-c the expected return from allow / sbrk have been updated.

Maybe this deserves more discussion.

Reply:
I am still having trouble deciding if this should appear in TRD 104. It can now no longer be returned on the platforms TRD 104 currently says it is for. Either TRD 104 needs to be about 64 bit / CHERI platforms, or actually this would need documenting in another document.

Reply:
That seems a good suggestion (with a note they are not possible on this platform, but reserved).